src/runners/baseRunner/BaseRunnerResultAnalyzer.ts

```diff
@@ -41,35 +42,61 @@ export abstract class BaseRunnerResultAnalyzer {
 
     public feedBack(): ITestResult[] {
         const result: ITestResult[] = [];
+        const itemMap: Map<string, ITestItem> = new Map<string, ITestItem>();
         for (const test of this.tests) {
-            this.processTestItemRecursively(test, result);
+            this.flatTestItems(test, itemMap);
         }
+        this.parseResults(result, itemMap);
+        this.addSkippedTests(result, itemMap);
         return result;
     }
 
-    protected processTestItemRecursively(testItem: ITestItem, resultList: ITestResult[]): void {
-        if (testItem.level === TestLevel.Method) {
-            resultList.push(this.processMethod(testItem));
-        } else {
-            testItem.children.forEach((child: ITestItem) => this.processTestItemRecursively(child, resultList));
+    protected abstract processData(data: string): void;
+
+    protected flatTestItems(item: ITestItem, map: Map<string, ITestItem>): void {
+        if (item.level === TestLevel.Method) {
+            map.set(item.fullName, item);
+        } else if (item.children) {
+            item.children.forEach((child: ITestItem) => this.flatTestItems(child, map));
         }
     }
 
-    protected abstract processData(data: string): void;
+    protected parseResults(resultArray: ITestResult[], itemMap: Map<string, ITestItem>): void {
+        for (const [key, value] of this.testResults.entries()) {
+            let result: ITestResult = Object.assign(_.cloneDeep(defaultResult), {
+                fullName: key,
+                details: value,
+            });
+
+            const item: ITestItem | undefined = itemMap.get(key);
+            if (item) {
+                result =  {
+                    displayName: item.displayName,
+                    uri: Uri.parse(item.uri).toString(),
+                    range: item.range,
+                    fullName: key,
+                    details: value,
+                };
+            } else {
+                result = Object.assign(result, {
+                    displayName: key.slice(key.indexOf('#') + 1),
+                });
+            }
 
-    protected processMethod(test: ITestItem): ITestResult {
-        let testResultDetails: ITestResultDetails | undefined = this.testResults.get(test.fullName);
-        if (!testResultDetails) {
-            testResultDetails = { status: TestStatus.Skip };
+            resultArray.push(result);
+            itemMap.delete(key);
         }
+    }
 
-        return {
-            displayName: test.displayName,
-            fullName: test.fullName,
-            uri: Uri.parse(test.uri).toString(),
-            range: test.range,
-            result: testResultDetails,
-        };
+    protected addSkippedTests(result: ITestResult[], itemMap: Map<string, ITestItem>): void {
+        for (const item of itemMap.values()) {
+            result.push(Object.assign(defaultResult, {
```

Comment:
value of `defaultResult` is changed on every execution of the line. 

Consider `Object.assign({}, defaultResult, { ... })` ?